backend/fms_core/template_importer/importers/sample_qc_spark.py

```diff
@@ -0,0 +1,127 @@
+    'Well positions': str,
+    '260nm': str,
+    '280nm': str,
+    '320nm': str,
+    'Concentration ug/ul': str,
+    'Purity 260/280': str,
+    'Mass/rxn (ug)': str
+})
+
+class SampleSparkQCImporter(GenericImporter):
+    SHEETS_INFO = SAMPLE_QC_SPARK_TEMPLATE["sheets info"]
+
+    def __init__(self):
+        super().__init__()
+        self.initialize_data_for_template()
+    
+    def preprocess_file(self, path: os.PathLike) -> StringIO:
+        new_content = StringIO()
+
+        with open(path) as original:
+            for line in original:
+                if line.startswith("Date of measurement"):
+                    break
+                new_content.write(line.strip())
+                new_content.write('\n')
+            for line in original:
+                if line.startswith("Plate ID: "):
+                    self.preloaded_data['plate_barcode'] = line[len("Plate ID: "):].strip()
+                    break
+            for line in original:
+                date_key = "Date: "
+                if line.startswith(date_key):
+                    self.preloaded_data["execution_date"] = line[len(date_key):len(date_key)+10].strip()
+
+        return new_content
+
+    def initialize_data_for_template(self):
+        #Get protocol for SampleQCSpark
+        protocol = Protocol.objects.get(name='Sample Quality Control Spark')
+
+        #Preload data
+        self.preloaded_data = {'process': None, 'protocol': protocol, 'process_properties': {}, 'plate_barcode': None}
+
+        self.preloaded_data['process'] = Process.objects.create(protocol=protocol,
+                                                                comment='Sample Quality Control (imported from template)')
+
+        # Preload PropertyType objects for the sample qc in a dictionary for faster access
+        try:
+            self.preloaded_data['process_properties'] = { property.name: {'property_type_obj': property } for property in
+                                                         list(PropertyType.objects.filter(object_id=protocol.id))}
+        except Exception as e:
+            self.base_errors.append(f'Property Type could not be found. {e}')
+
+    def import_template_inner(self):
+        spark_sample_qc_sheet = self.sheets['SampleQCSpark']
+
+        # TODO: handle None?
+        candidate_step = Step.objects.filter(protocol=self.preloaded_data['protocol']).first()
+
+        # Add the template to the process
+        if self.imported_file is not None:
+            self.preloaded_data['process'].imported_template_id = self.imported_file.id
+            self.preloaded_data['process'].save()
+
+        for row_id, row_data in enumerate(spark_sample_qc_sheet.rows):
+            row_data: SampleQCSparkSheet = row_data
+            process_measurement_properties = self.preloaded_data['process_properties']
+
+            quantity_flag = str_cast_and_normalize(float(row_data['Mass/rxn (ug)']) > 1000)
```

Comment:
Don't convert to float if you don't have to the more conversion we do the more error we introduce. Most calculations are done using decimals.